relay-server/src/actors/envelopes.rs

```diff
@@ -449,199 +447,53 @@ impl Handler<QueueEnvelope> for EnvelopeManager {
-                                    envelope_context.reject(outcome);
-                                    ProcessingError::ScheduleFailed
-                                }
-
-                                #[cfg(feature = "processing")]
-                                SendEnvelopeError::StoreFailed(e) => {
-                                    envelope_context.reject(outcome);
-                                    ProcessingError::StoreFailed(e)
-                                }
-
-                                SendEnvelopeError::BodyEncodingFailed(e) => {
-                                    envelope_context.reject(outcome);
-                                    ProcessingError::BodyEncodingFailed(e)
-                                }
-
-                                SendEnvelopeError::EnvelopeBuildFailed(e) => {
-                                    envelope_context.reject(outcome);
-                                    ProcessingError::EnvelopeBuildFailed(e)
-                                }
-
-                                SendEnvelopeError::UpstreamRequestFailed(e) => {
-                                    if e.is_received() {
-                                        envelope_context.accept();
-                                    } else {
-                                        envelope_context.reject(outcome);
-                                    }
-
-                                    ProcessingError::UpstreamRequestFailed(e)
-                                }
-                            })
-                        }
-                    })
-                    .into_actor(slf)
-            })
-            .map_err(move |error, slf, context| {
-                if Capture::should_capture(&slf.config) {
-                    context.notify(Capture::rejected(event_id, &error));
+                Err(SendEnvelopeError::UpstreamRequestFailed(e)) if e.is_received() => {
+                    envelope_context.accept();
+                    Ok(())
                 }
-
-                let outcome = error.to_outcome();
-                if let Some(Outcome::Invalid(DiscardReason::Internal)) = outcome {
+                Err(error) => {
                     // Errors are only logged for what we consider an internal discard reason. These
-                    // indicate errors in the infrastructure or implementation bugs. In other cases,
-                    // we "expect" errors and log them as debug level.
+                    // indicate errors in the infrastructure or implementation bugs.
                     relay_log::with_scope(
                         |scope| scope.set_tag("project_key", project_key),
-                        || relay_log::error!("error processing envelope: {}", LogError(&error)),
+                        || relay_log::error!("error sending envelope: {}", LogError(&error)),
                     );
-                } else {
-                    relay_log::debug!("dropped envelope: {}", LogError(&error));
```

Comment:
All debug logs are now in the `EnvelopeContext`. Instead of logging the error, we simply log the outcome, since it contains about the same information. The same goes for `Capture::reject` below, since capture mode doesn't need more information. All this was changed to greatly simplify error handling.

In contrast, hard errors (bugs) are still logged to Sentry where they can be caught.